news/12285-improve-plugin-subcommand-arg-parsing

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Refactor subcommand argument parsing to make it easier to understand (#12285)
```

Comment:
This should add a mention of the change in when plugin subcommands are being called, earlier in the stack.